testing/test_terminal.py

```diff
@@ -2537,6 +2614,27 @@ def test_format_session_duration(seconds, expected):
     assert format_session_duration(seconds) == expected
 
 
+@pytest.mark.parametrize(
+    "seconds, expected",
+    [
+        (3600 * 100 - 60, "99h 59m"),
+        (31 * 60 - 1, "30m 59s"),
+        (10.1236, "10.124s"),
+        (9.1236, "9.124s"),
+        (0.1236, "123.6ms"),
+        (0.01236, "12.36ms"),
+        (0.001236, "1.236ms"),
+        (0.0001236, "123.6us"),
+        (0.00001236, "12.36us"),
+        (0.000001236, "1.236us"),
+    ],
+)
+def test_format_node_duration(seconds, expected):
+    from _pytest.terminal import format_node_duration
+
+    assert format_node_duration(seconds) == expected
```

Comment:
Nice testing this in a unittest!

```suggestion
def test_format_node_duration(seconds: float, expected: str) -> None:
    from _pytest.terminal import format_node_duration

    assert format_node_duration(seconds) == expected
```